src/query.cpp

```diff
@@ -132,115 +132,90 @@ Rcpp::NumericVector transform_timestamp_points(
 }
 
 Rcpp::DataFrame transform_rows(qdb_point_result_t ** rows,
-    qdb_size_t rows_count,
-    qdb_size_t columns_count,
+    qdb_size_t row_count,
+    qdb_size_t column_count,
     const Rcpp::StringVector & columns)
 {
     assert(rows);
-    assert(rows_count > 0);
-    assert(columns_count > 0);
+    assert(column_count > 0);
 
-    Rcpp::List df; // {rows_count * columns_count};
+    Rcpp::DataFrame df; // {row_count * column_count};
 
-    for (qdb_size_t column_index = 0u; column_index < columns_count;
-         ++column_index)
+    if (row_count)
     {
-        switch (rows[0][column_index].type)
+        for (qdb_size_t column_index = 0u; column_index < column_count;
+             ++column_index)
         {
-        case qdb_query_result_blob:
-            df.push_back(transform_blob_points(rows, rows_count, column_index));
-            break;
-
-        case qdb_query_result_count:
-            df.push_back(
-                transform_count_points(rows, rows_count, column_index));
-            break;
-
-        case qdb_query_result_double:
-            df.push_back(
-                transform_double_points(rows, rows_count, column_index));
-            break;
-
-        case qdb_query_result_int64:
-            df.push_back(
-                transform_int64_points(rows, rows_count, column_index));
-            break;
-
-        case qdb_query_result_timestamp:
-            df.push_back(
-                transform_timestamp_points(rows, rows_count, column_index));
-            break;
-
-        default:
-            assert(!"unknown data type");
-            df.push_back(Rcpp::GenericVector(rows_count));
-            break;
+            const std::string column_name =
```

Comment:
Make non-const and move in `push_back()` below.

Reply:
Yes, should have changed that back, I was battling to figure what kind of type was needed for dataframe push back.